lib/smack/Regions.cpp

```diff
@@ -143,15 +101,22 @@ void Region::init(const Value *V, unsigned length) {
   }
 
   singleton =
-      DL && representative && isSingleton(representative, offset, length);
+      DL && representative && isSingleton(V, length);
   allocated = !representative || isAllocated(representative);
+  // Shaobo: I removed the dependency of the bytewise flag on isFieldDisjoint
```

Comment:
It would be good to clean this up and figure it out.

Reply:
This comment is not up-to-date. I removed it. I think I have figured out it.